src/components/NavBar/NavBar.tsx

```diff
@@ -19,25 +20,43 @@ const useStyles = makeStyles<Theme>(theme => ({
 
 const NavBar: React.FC<NavBarProps> = props => {
   const classes = useStyles();
+  const [isMenuVisible, updateMenuVisibility] = useState(false);
 
   const button: NavBarButton = {
     menu: () => <MenuIcon />,
     back: () => <ArrowBackIcon />,
   };
   const { type, onClick } = props;
 
+  const showSlideMenu = () => {
+    updateMenuVisibility(!isMenuVisible);
+  };
+
+  const handleOnClick = () => {
+    onClick ? onClick() : showSlideMenu();
```

Comment:
https://github.com/BuildIt-Poland/buddy-app/pull/37/files#diff-d414628bfd8fb9e200ec63cc1a483446L49

You deleted `onClick` props but it still required in the navBar component. Can you explain us this line?

Reply:
It's not. `onClick` is optional prop.
You can preview it here: https://github.com/BuildIt-Poland/buddy-app/blob/fcdabc4d49685abb1895ea68c03909f304e6454c/src/components/NavBar/types.ts#L5

You can pass callback function to handle user redirection to previous screen when using Navbar with back icon (not hamburger)

```
      <NavBar
        type='back'
        onClick={() =>
          props.history.push(ROUTES.BUDDY_TASKS_LIST.replace(':newbieId', newbieId))
        }
      />
```